core_contracts/governance/governance.py

```diff
@@ -75,8 +89,56 @@ def launchBalanced(self) -> None:
     @external
     @only_owner
     @payable
-    def createAndFundMarket(self, _market_name: str) -> None:
-        pass
+    def createBnusdMarket(self) -> None:
+        value = self.msg.value
+        if value == 0:
+            revert(f'ICX sent must be greater than zero.')
+        dex_address = self.addresses['dex']
+        sICX_address = self.addresses['sicx']
+        bnUSD_address = self.addresses['bnUSD']
+        staking = self.create_interface_score(self.addresses['staking'], StakingInterface)
+        rewards = self.create_interface_score(self.addresses['rewards'], RewardsInterface)
+        loans = self.create_interface_score(self.addresses['loans'], LoansInterface)
+        bnUSD = self.create_interface_score(bnUSD_address, AssetInterface)
+        sICX = self.create_interface_score(sICX_address, AssetInterface)
+        dex = self.create_interface_score(dex_address, DexInterface)
+        price = bnUSD.priceInLoop()
+        amount = UNITS_PER_TOKEN * value // (price * 7)
+        staking.icx(value // 7).stakeICX()
+        loans.icx(self.icx.get_balance(self.address)).depositAndBorrow('bnUSD', amount)
+        bnUSD_value = bnUSD.balanceOf(self.address)
+        sICX_value = sICX.balanceOf(self.address)
+        bnUSD.transfer(dex_address, bnUSD_value, json_dumps({"method": "_deposit"}).encode())
+        sICX.transfer(dex_address, sICX_value, json_dumps({"method": "_deposit"}).encode())
+        dex.add(sICX_address, bnUSD_address, sICX_value, bnUSD_value)
+        name = 'sICX/bnUSD'
+        pid = 2
```

Comment:
Instead, use the function:

getPoolId(self, _token1Address: Address, _token2Address: Address)
passing in the addresses of sICX and bnUSD

Reply:
Nice. Thanks.

Reply:
Should be corrected now.